File: service/src/test/java/coding/challenge/DataLoader.java
package coding.challenge;

import java.net.URISyntaxException;
import java.nio.file.Path;

public final class DataLoader {

    private DataLoader() {
    }

    public static Path loadFile(String filename) {
        try {
            return Path.of(DataLoader.class.getClassLoader().getResource(filename).toURI());
        } catch (URISyntaxException e) {
            throw new RuntimeException(e);
        }
    }

}

File: service-kt/src/test/kotlin/coding/challenge/DataLoader.kt
package coding.challenge

import java.net.URISyntaxException
import java.nio.file.Path

object DataLoader {

    fun loadFile(filename: String): Path {
        try {
            return Path.of(this.javaClass.classLoader.getResource(filename)?.toURI())
        } catch (e: URISyntaxException) {
            throw RuntimeException(e)
        }
    }

}
